src/components/page/page_template.tsx

```diff
@@ -37,47 +39,65 @@ export const TEMPLATES = [
   'empty',
 ] as const;
 
-export type EuiPageTemplateProps = Omit<EuiPageProps, 'paddingSize'> & {
-  /**
-   * Choose between 3 types of templates.
-   * `default`: Typical layout with nothing centered
-   * `centeredBody`: The panelled content is centered
-   * `centeredContent`: The content inside the panel is centered
-   * `empty`: Removes the panneling of the page content
-   */
-  template?: typeof TEMPLATES[number];
-  /**
-   *
-   * Padding size will not get applie to the over-arching #EuiPage,
-   * but will propogate through all the components to keep them in sync
-   */
-  paddingSize?: typeof SIZES[number];
-  /**
-   * Optionally include #EuiPageSideBar content.
-   * The inclusion of this will affect the whole layout
-   */
-  pageSideBar?: ReactNode;
-  /**
-   * Gets passed along to the #EuiPageSideBar component
-   */
-  pageSideBarProps?: EuiPageSideBarProps;
-  /**
-   * Optionally include an #EuiPageHeader by passing an object of its props
-   */
-  pageHeader?: EuiPageHeaderProps;
-  /**
-   * Gets passed along to the #EuiPageBody component
-   */
-  pageBodyProps?: EuiPageBodyProps;
-  /**
-   * Gets passed along to the #EuiPageContent component
-   */
-  pageContentProps?: EuiPageContentProps;
-  /**
-   * Gets passed along to the #EuiPageContentBody component
-   */
-  pageContentBodyProps?: EuiPageContentBodyProps;
-};
+type _EuiPageTemplateTypes = ExclusiveUnion<
+  {
+    template?: 'default';
+    /**
+     * Adds contents inside of an EuiBottomBar.
+     * Only works when `template = 'default'`
+     */
+    bottomBar?: EuiBottomBarProps['children'];
+    /**
+     * Gets passed along to the #EuiBottomBar component if `bottomBar` has contents
+     */
+    bottomBarProps?: EuiBottomBarProps;
+  },
+  {
+    /**
+     * Choose between 3 types of templates.
+     * `default`: Typical layout with nothing centered
+     * `centeredBody`: The panelled content is centered
+     * `centeredContent`: The content inside the panel is centered
+     * `empty`: Removes the panneling of the page content
+     */
+    template?: typeof TEMPLATES[number];
```

Comment:
```suggestion
    template: typeof TEMPLATES[number];
```

as it is no longer the default case